bulbs/poll/models.py

```diff
@@ -0,0 +1,94 @@
+from bulbs.content.models import Content
+from bulbs.utils import vault
+from django.db import models
+from djes.models import Indexable
+from rest_framework.exceptions import APIException
+
+from time import mktime
+import pytz
+import requests
+
+SODAHEAD_DATE_FORMAT = '%m/%d/%y %I:%M %p'
+
+class Poll(Content):
+    class SodaheadResponseError(APIException):
+        status_code = 503
+        default_detail = "Third-party poll provider temporarily unavailable."
+
+    question_text = models.TextField(blank=True, default="")
+    sodahead_id = models.CharField(max_length=20, blank=True, default="")
+    last_answer_index = models.IntegerField(default=0)
+    end_date = models.DateTimeField(null=True, default=None)
+
+    def sodahead_payload(self):
+        poll_payload = {
+            'access_token': vault.read('sodahead/token'),
+            'id': self.sodahead_id,
+            'name': self.title,
+            'title': self.question_text,
+        }
+
+        if self.published:
+            activation_date = self.published.astimezone(pytz.utc)
+            poll_payload['activationDate'] = activation_date.strftime(SODAHEAD_DATE_FORMAT)
+
+        if self.end_date:
+            end_date = self.end_date.astimezone(pytz.utc)
+            poll_payload['endDate'] = end_date.strftime(SODAHEAD_DATE_FORMAT)
+
+        for answer in self.answers.all():
+            if answer.answer_text is u'':
+                poll_payload[answer.sodahead_answer_id] = 'Intentionally blank'
+            else:
+                poll_payload[answer.sodahead_answer_id] = answer.answer_text
+
+        if not 'answer_01' in poll_payload:
+            poll_payload['answer_01'] = 'default answer 1'
+
+        if not 'answer_02' in poll_payload:
+            poll_payload['answer_02'] = 'default answer 2'
+
+        return poll_payload
+
+    def save(self, *args, **kwargs):
+        if self.sodahead_id is "":
```

Comment:
Shouldn't use "is" for comparing to empty string -- "is" is for comparing identity, "==" is for comparing value. 

Use "==", or better yet just "not" since it's ID is a CharField:      if not self.sodahead_id